### PR TITLE
Feature/adapt for ev ami

### PR DESCRIPTION
**Pull request by mikezhong-embark**
Updated the service to work with EV specific AMIs, tested on instance launched using an extended `asg-prod-pipeline` launch template

### CODE REVIEW

File: config/ebs-autoscale.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
       "logical_volume": "autoscale_lv"
     },
     "volume": {
-        "type": "gp2",
+        "type": "gp3",
```
Comment (mattbarton) [inline config/ebs-autoscale.json:9]: We're still using gp2 in production AFAIK until someone tackles this card: https://trello.com/c/HCT87tb0/1816-investigate-migrate-ebs-gp2-to-gp3

One risk of switching to gp3 before we've tackled that card is that we've increased our gp2 AWS limits, but have not yet increased our gp3 limits, AFAIK

File: bin/create-ebs-volume [hunk 2]
```diff
@@ -143,14 +143,31 @@ if [[ ! "$SIZE" ]]; then
...
+declare -A device_name_map
+declare -a device_ordering
+
+# associative arrays do not care for your logical ordering so keep order here
+device_ordering=( '/dev/nvme1n1' '/dev/nvme2n1' '/dev/nvme3n1' '/dev/nvme4n1' '/dev/nvme5n1' '/dev/nvme6n1' )
+
+# mapping nvme device names to xvdb 
+device_name_map=( ['/dev/nvme1n1']='/dev/xvdba' ['/dev/nvme2n1']='/dev/xvdbb' ['/dev/nvme3n1']='/dev/xvdbc' ['/dev/nvme4n1']='/dev/xvdbd' ['/dev/nvme5n1']='/dev/xvdbe' ['/dev/nvme6n1']='/dev/xvdbf')
+
+# alphabet=( {a..z} )
```
Comment (mattbarton) [inline bin/create-ebs-volume:157]: Should remove commented out code

File: bin/create-ebs-volume [hunk 3]
```diff
@@ -143,14 +143,31 @@ if [[ ! "$SIZE" ]]; then
...
-alphabet=( {a..z} )
+# -- Mike Zhong -- #
+
+declare -A device_name_map
+declare -a device_ordering
+
+# associative arrays do not care for your logical ordering so keep order here
+device_ordering=( '/dev/nvme1n1' '/dev/nvme2n1' '/dev/nvme3n1' '/dev/nvme4n1' '/dev/nvme5n1' '/dev/nvme6n1' )
+
+# mapping nvme device names to xvdb
```
Comment (mattbarton) [inline bin/create-ebs-volume:154]: In general we prefer comments that say 'why' (or sometimes 'how') rather than 'what'

**Review by mattbarton — APPROVED** [review]
Just one substantive comment concerning whether we're ready to use gp3 in production yet

File: config/ebs-autoscale.json (hunk 1 above)
Comment (mikezhong-embark) [reply config/ebs-autoscale.json:9]: I see, happy to revert back to `gp2`, will need to make this update in the `autoscaling_cloudformation.yaml` file as well. I saw the card and wasn't sure if the migration was intended to be gradual or all at once. As it stands, the root volume is still `gp2` and 256GB, so any disk usage outside of `/tmp` would be gp2

File: bin/create-ebs-volume (hunk 2 above)
Comment (mikezhong-embark) [reply bin/create-ebs-volume:157]: done

File: bin/create-ebs-volume (hunk 3 above)
Comment (mikezhong-embark) [reply bin/create-ebs-volume:154]: added the why

**Pull request merged**